File: iosApp/iosApp/DesignSystem/Motion.swift
import SwiftUI

/// 모션. 안드로이드 `TeamFisMotion` 과 값이 같아야 한다 — 한쪽만 고치지 않는다.
///
/// 감속 위주 이징 하나만 쓴다. 시간으로 성격을 구분한다.
enum TeamFisMotion {
    /// 눌림, 토글
    static let fast = Animation.timingCurve(0.2, 0, 0, 1, duration: 0.12)

    /// 선택 이동, 페이드
    static let base = Animation.timingCurve(0.2, 0, 0, 1, duration: 0.2)

    /// 접힘·펼침, 화면 전환
    static let slow = Animation.timingCurve(0.2, 0, 0, 1, duration: 0.32)
}

File: androidApp/src/main/kotlin/com/teamfis/app/ui/theme/Motion.kt
package com.teamfis.app.ui.theme

import androidx.compose.animation.core.CubicBezierEasing
import androidx.compose.animation.core.TweenSpec
import androidx.compose.animation.core.tween

/**
 * 모션. iOS `TeamFisMotion` 과 값이 같아야 한다 — 한쪽만 고치지 않는다.
 *
 * 감속 위주 이징 하나만 쓴다. 시간으로 성격을 구분한다.
 */
object TeamFisMotion {
    val easing = CubicBezierEasing(0.2f, 0f, 0f, 1f)

    /** 눌림, 토글 */
    fun <T> fast(): TweenSpec<T> = tween(durationMillis = 120, easing = easing)

    /** 선택 이동, 페이드 */
    fun <T> base(): TweenSpec<T> = tween(durationMillis = 200, easing = easing)

    /** 접힘·펼침, 화면 전환 */
    fun <T> slow(): TweenSpec<T> = tween(durationMillis = 320, easing = easing)
}
